hibernate-core/src/main/java/org/hibernate/query/sql/internal/ParameterParser.java

```diff
@@ -131,6 +131,11 @@ else if ( '\\' == c ) {
 				if ( c == ':' && indx < stringLength - 1 && sqlString.charAt( indx + 1 ) == ':') {
```

Comment:
I was wondering why we were even attempting to do this sort of escaping and had a look at https://hibernate.atlassian.net/browse/HHH-1237 which introduced this. I see that @gavinking agrees with my view on this, which is that this is wrong.
Deciding if a `:` is the start token for a named parameter should IMO be done differently and there should be no need for escaping. If we assume people can alter their SQL so it can be interpreted by Hibernate ORM, then I would prefer to tell them to insert a whitespace before/after a `:` to disambiguate a parameter from a pass-through character, if necessary at all.

According to the [Jakarta Persistence spec](https://jakarta.ee/specifications/persistence/3.1/jakarta-persistence-spec-3.1#a4401), the parameter name must follow the [Java identifier syntax rules](https://jakarta.ee/specifications/persistence/3.1/jakarta-persistence-spec-3.1#a4760). So if `!Character.isJavaIdentifierStart( sqlString.charAt( indx + 1 ) )`, then we just pass-through all following colon characters.

Reply:
Yeah, I was pretty upset to discover that we patched the native SQL parser in a way that allowed the rather uncommon `:=` syntax, at the cost of breaking the rather more common `::` syntax in Postgres. I'm amazed that people didn't _scream_ when that change was made. It just doesn't look like it was properly thought through at all.

The thing is that there's a totally reasonable way to do this that would have accepted _both_ syntaxes without introducing weird new escaping syntax:

- detect `::` or `:=` or colon followed by any other symbols and leave it alone
- only interpret `:name` as a named parameter _when not preceded by a colon_

But what do we do about it now? Removing this escaping would break all the queries of people who have had to work around the original bogus solution.

Reply:
You argued that this behavior is undocumented and as such can be considered a bug, so IMO we should just fix this bug in ORM 6.5.

Reply:
Well OK I don't object to doing that.

Reply:
I updated additional commit.
I have changed my approach. `::=` is not common sql syntax but released code are accepted it. so added the preprocessor for like this none-common case and removed the codes for complexed esacaping.
But previous code too didn't recognized `::` (type casting token). So I updated the condition for filtering name param token also.

I am waiting for your opinions. If you are thinking to change it as refactoring, not simple patch.  I also agree it.